### PR TITLE
Makefile: Split `make test` into two test parts for CI

### PR DESCRIPTION
**Pull request by qrr1995**
### What problem does this PR solve? 
We have two CI  jobs include `tidb_ghpr_check` which runs `make checklist check explaintest` and `tidb_ghpr_check_2`   which runs `make checkdep gotest `.

The explicitly `make` command leads to compatibility problem when we add `make gogenerate` for `master` branch since `gogenerate` rule doesn't exist in `release-` branch.

### What is changed and how it works?

Split the tests into two test chunks , and jenkins CI would run check tests implicitly by
 `make check_test_chunk_1` 
and 
`make check_test_chunk_2`

### CODE REVIEW

**Comment by Deardrops** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -126,7 +126,11 @@ clean:
 	rm -rf *.out
 	rm -rf parser
 
-test: checklist checkdep gotest explaintest gogenerate
+test: test_part_1 test_part_2
```
Comment (SunRunAway) [inline Makefile]: We could add an annotation here explaining why it should be splited into 2 parts.

**Comment by qrr1995** [comment]
Do not cherry pick to release branches anymore since `tidb_ghpr_check` would `make dev`  for them now.

**Pull request merged**